pymongo/topology_description.py

```diff
@@ -447,9 +447,21 @@ def _updated_topology_description_srv_polling(topology_description, seedlist):
         return topology_description
 
     # Add SDs corresponding to servers recently added to the SRV record.
-    for address in seedlist:
-        if address not in sds:
+    if topology_description._srv_max_hosts != 0:
+        available_new_hosts = set(seedlist) - set(sds.keys())
+        new_hosts_needed = topology_description._srv_max_hosts - len(sds)
+        if new_hosts_needed > 0:
+            seedlist = random.sample(available_new_hosts, min(new_hosts_needed,
+                                                              len(available_new_hosts)))
+        elif new_hosts_needed < 0:
```

Comment:
Since srv_max_hosts is set once and never changes, shouldn't it be impossible to have more than that many hosts? If so this is dead code that we can just remove.

Reply:
srv_max_hosts never changes but I don't understand how that means that it's impossible to have more than that number of hosts returned. This is the code section that enforces the number of hosts. Seedlist should just be every discovered seed host right? There's nothing about the seedlist that enforces a limit on the number of hosts that it gets from _SrvResolver, we have to impose that limit ourselves by popping off seeds or adding seeds as necessary. I might have misunderstood, though.